src/modules/samples/rust/src/lib.rs

```diff
@@ -0,0 +1,136 @@
+use libc::{c_int, c_char ,c_void};//, c_void
+use std::ffi::CStr;
+use std::ffi::CString;
+use std::ptr::NonNull;
+
+mod sample;
+
+// Constants and Method headers were generated by bindgen
+// (https://github.com/rust-lang/rust-bindgen)
+pub const MMI_OK: u32 = 0;
+pub type MMI_HANDLE = *mut ::std::os::raw::c_void;
+pub type MMI_JSON_STRING = *mut ::std::os::raw::c_char;
+
+// MSFT change: Added the #[no_mangle] annotation and defined the methods
+#[no_mangle]
+pub extern "C" fn MmiGetInfo(
+        clientName: *const ::std::os::raw::c_char,
+        payload: *mut MMI_JSON_STRING,
+        payloadSizeBytes: *mut ::std::os::raw::c_int,
+    ) -> ::std::os::raw::c_int {
+    let c_str: &CStr = unsafe { CStr::from_ptr(clientName) };
+    let str_slice: &str = c_str.to_str().unwrap();
+    println!("{}",str_slice);
+    // Dereferencing a pointer
+    unsafe{*payloadSizeBytes = 5};
+    // Converting a string to a c_char pointer
+    unsafe{let temppayload = CString::new("newpayload").unwrap();
+    *payload= CString::into_raw(temppayload);}
+    // Dummy return
+    return 0;
+}
+
+#[no_mangle]
+pub extern "C" fn MmiOpen(
+        clientName: *const ::std::os::raw::c_char,
```

Comment:
Please use `snake_case` for variable/argument names (https://rust-lang.github.io/api-guidelines/naming.html)